script.rb

```diff
@@ -0,0 +1,172 @@
+# rubocop:disable all
+
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    var = to_a
+    return to_enum(:my_each) unless block_given?
+
+    index = 0
+    while index < var.length
+      yield(var[index])
+      index += 1
+    end
+    var
+  end
+
+  def my_each_with_index
+    var = to_a
+    return to_enum(:my_each) unless block_given?
+
+    index = 0
+    while index < var.length
+      yield(var[index], index)
+      index += 1
+    end
+    var
+  end
+
+  def my_select
+    var = to_a
+    new_arr = []
+    return to_enum(:my_select) unless block_given?
+
+    var.my_each do |x|
+      new_arr.push(x) if yield(x) == true
+    end
+    new_arr
+  end
+
+  def my_all?(pattern = nil)
+    var = to_a
+    if block_given? == true
+      var.my_select { |x| yield(x) }.length == var.length
+    elsif !pattern.nil?
+      if pattern.class == Regexp
+        var.my_select { |x| !pattern.match(x).nil? }.length == var.length
+      elsif pattern.class == Integer
+        var.my_select { |x| x == pattern }.length == var.length
+      else
+        var.my_select { |x| x.class == pattern }.length == var.length
+      end
+    else
+      var.my_select { |x| !x.nil? && x != false }.length == var.length
+    end
+  end
```

Comment:
Here, the tester shows:

![image](https://user-images.githubusercontent.com/2739245/94213848-9a48bc00-fe9d-11ea-866e-6406baa3b121.png)

Perhaps your code isn't processing for Classes? Please make sure your code can process classes, as well as other types of data.